src/main/java/com/autonomic/tmc/auth/exception/BaseSdkException.java

```diff
@@ -55,30 +55,35 @@ static String buildMessage(ErrorSourceType errorSourceType, String clientMessage
     static class ProjectProperties {
 
         private static final String UNKNOWN = "[ UNKNOWN ]";
-        static String pom = "pom.xml";
+        private Manifest manifest;
 
-        private Model properties;
-
-        public ProjectProperties() {
-            try {
-                MavenXpp3Reader reader = new MavenXpp3Reader();
-                properties = reader.read(new FileReader(pom));
-            } catch (IOException | XmlPullParserException e) {
-                log.warn("Failed to read pom.xml while building BaseSdkException", e);
-                properties = new Model();
+        private Manifest getManifest() {
+            if (Objects.isNull(manifest)) {
+                try {
+                    String jarPath = BaseSdkException.class.getProtectionDomain()
+                        .getCodeSource().getLocation().toURI().getPath();
+                    try (JarFile jarFile = new JarFile(jarPath)) {
+                        manifest = jarFile.getManifest();
+                    }
+                } catch (IOException | URISyntaxException ignored) {}
             }
+            return manifest;
         }
 
-        String getName() {
-            return Optional.ofNullable(properties.getArtifactId()).orElse(UNKNOWN);
+        private String getAttribute(String name) {
+            return Optional.ofNullable(getManifest())
+                .map(m -> m.getMainAttributes().getValue(name))
+                .orElse(null);
         }
 
-        String getVersion() {
-            return Optional.ofNullable(properties.getVersion()).orElse(UNKNOWN);
+        String getName() {
+            return Optional.ofNullable(getAttribute("Implementation-Title"))
+                .orElse(UNKNOWN);
         }
 
-        void setProperties(Model properties) {
-            this.properties = properties;
+        String getVersion() {
+            return Optional.ofNullable(getAttribute("Implementation-Version"))
+                .orElse(UNKNOWN);
```

Comment:
```suggestion
            return getAttribute("Implementation-Version");
```
See other suggestions for complete picture.